M2Mqtt/Messages/MqttMsgPublish.cs

```diff
@@ -30,18 +31,55 @@ public class MqttMsgPublish : MqttMsgBase
         /// Message topic
         /// </summary>
         public string Topic { get; set; }
-        
+
         /// <summary>
         /// Message data
         /// </summary>
         public byte[] Message { get; set; }
 
+        /// <summary>
+        /// True if the payload is UTF8 encoded, v5.0 only
+        /// </summary>
+        public bool IsPayloadUTF8 { get; set; }
+
+        /// <summary>
+        /// Message Expiry Interval. If the server did not managed to process it on time, the message must be deleted, v5.0 only
+        /// </summary>
+        public uint MessageExpiryInterval { get; set; }
```

Comment:
Comment and/or name should be clear (or clarify) what's the unit. I suspect it's milliseconds.

If that's the case, this shouldn't be unsigned because -1 it's the infinite timeout constant

Reply:
The default value has to be 0 to match the MQTT specification, if anything else, it' won't work for serialization and deserialization.
For the units, those are seconds. I'll add more elements in the description.